src/main/java/seedu/address/logic/parser/IndexIdPair.java

```diff
@@ -0,0 +1,129 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_LOCATION_ID;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PERSON_ID;
+
+import java.util.List;
+import java.util.Optional;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.Model;
+import seedu.address.model.attribute.Id;
+import seedu.address.model.location.Location;
+import seedu.address.model.person.Person;
+
+/**
+ * This represents an identifier that can identify a Person or a Location.
+ * The identifier may be using either an Id or an index from user input.
+ */
+public class IndexIdPair extends ReadOnlyIndexIdPair {
+
+    public static final String MESSAGE_INVALID_PERSON_COMMAND_USE = "This pair refers to a location, not a person.";
+    public static final String MESSAGE_INVALID_LOCATION_COMMAND_USE = "This pair refers to a person, not a location.";
+    public static final String MESSAGE_NO_ID_NOR_INDEX = "User input has neither Id nor index.";
+
+    private Prefix prefix;
+
+    /**
+     * Creates an IndexIdPair that can identify either a person or a location.
+     * The {@code ArgumentMultimap} should contain either index or Id user input but not both.
+     * @param argMultimap contains the Id or index user input.
+     * @param prefix determines if this IndexIdPair identifies a person or a location.
+     * @throws ParseException if unable to parse the Id or the index.
+     */
+    public IndexIdPair(ArgumentMultimap argMultimap, Prefix prefix) throws ParseException {
+        requireAllNonNull(argMultimap, prefix);
```

Comment:
As this is a public constructor, perhaps you could consider making this code more defensive by ensuring that Prefix could only be PREFIX_PERSON_ID or PREFIX_LOCATION_ID? Since a wrong argument is likely to originate from a command inputting the wrong prefix, an assert statement here should be good enough.

Reply:
Alternatively, checking and throwing an exception may make this constructor easier to test.

Reply:
Sure! I have added exception checking.